src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ");
+        builder.append(dateFrom).append(" - ").append(dateTo).append(System.lineSeparator());
+        LocalDate parseDateFrom = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate parseDateTo = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        for (int i = 0; i < names.length; i++) {
+            builder.append(names[i]).append(" - ");
+            int salary = 0;
+            for (String dataElement : data) {
+                String[] arrayOfDataElements = dataElement.split(" ");
+                LocalDate parseParticularDate = LocalDate.parse(arrayOfDataElements[0],
```

Comment:
```suggestion
                LocalDate parseParticularDate = LocalDate.parse(arrayOfDataElements[DATE_INDEX],
```
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/08675068-e38f-486c-97f4-35a9ccffe45a)